src/main/java/org/broadinstitute/hellbender/utils/MathUtils.java

```diff
@@ -44,12 +59,408 @@
+            if (nextValue < minValue) {
+                minValue = nextValue;
+                minIndex = i;
+            }
+        }
+        return minIndex;
+    }
+
+    /**
+     * Given an int array returns the difference between the second smallest element
+     * and the smallest element.
+     * @param values the input values.
+     * @param defaultValue value to return in case the input array has zero or only one element.
+     * @return 0 if the input array has less than 2 elements, otherwise the second smallest
+     * - the smallest.
+     * @throws IllegalArgumentException if {@code values} is {@code null}.
+     */
+    public static int secondSmallestMinusSmallest(final int[] values, final int defaultValue) {
+        Utils.nonNull(values);
+        if (values.length <= 1) {
+            return defaultValue;
+        } else {
+            int smallest = values[0];
+            int secondSmallest = Integer.MAX_VALUE;
+            for (int i = 1; i < values.length; i++) {
+                if (values[i] < smallest) {
+                    secondSmallest = smallest;
+                    smallest = values[i];
+                } else if (values[i] < secondSmallest) {
+                    secondSmallest = values[i];
+                }
+            }
+            return secondSmallest - smallest;
+        }
+    }
+
+    /**
+     * Returns the smallest power of 2 that exceeds or equals {@code val}
+     * @param val input value
+     * @return power of 2 integer
+     * @throws IllegalArgumentException if an int overlow is encountered
+     */
+    public static int smallestPowerOfTwoGreaterThan(final int val)
+            throws IllegalArgumentException {
+        if (val > Integer.MAX_VALUE/2 || val < 0)
+            throw new IllegalArgumentException("The mallest power of 2 greater than " + val + " is greater than Integer.MAX_VALUE" +
+                    " or negative input.");
+        return val > 1 ? Integer.highestOneBit(val - 1) << 1 : 1;
+    }
+
+    /**
+     * Shrinks or expands a double array uniformly to a given length >= 2 ({@code newLength}) using nearest
+     * neighbor interpolation. The routine works as follows: it puts the array on a uniform partition of the unit
+     * interval on the real axis and and overlaps it with the desired grid. The value at a new grid point is
+     * set to its nearest neighbor from the original grid. For example, if data = {1, 2, 3} and newLength = 4,
+     * the output will be {1, 2, 2, 3}:
+     *
+     *      data     = 1     2     3
+     *      old grid = x     x     x
+     *      new grid = x   x   x   x
+     *      new data = 1   2   2   3
+     *
+     * @param data original array
+     * @param newLength length of the new array
+     * @return interpolated array
+     * @throws IllegalArgumentException if the input array is empty
+     */
+    public static double[] nearestNeighborUniform1DInterpolate(@Nonnull final double[] data, final int newLength)
```

Comment:
Remove `throws` and perform null checks.

Reply:
done done